### PR TITLE
refactor(commands): output `stderr` in `:sh` popup

### PR DESCRIPTION
**Pull request by RoloEdits**
Rather than getting the error in the one row of the status bar, and needing to check the logs to see what the error actually was, it now outputs to the popup. This doesn't address every use case, but for when it's sufficient, its a very nice quality of life improvement. Other options could come later to improve this, like what was said [here](https://github.com/helix-editor/helix/issues/2700#issuecomment-1206328317)
> A horizontal split with a scratch buffer similar to what log does could be a good approach

But I don't see a reason not to have a quick fix for something now.

sh_stderr_output

Connects: #2700

### CODE REVIEW

File: helix-term/src/commands.rs [hunk 1]
```diff
@@ -5701,27 +5701,29 @@ async fn shell_impl_async(
         process.wait_with_output().await?
     };
 
-    if !output.status.success() {
-        if !output.stderr.is_empty() {
-            let err = String::from_utf8_lossy(&output.stderr).to_string();
```
Comment (archseer) [inline helix-term/src/commands.rs:5706]: Why not use the lossy function like we did before? That way at least something is displayed

Comment (RoloEdits) [reply helix-term/src/commands.rs:5706]: The thought was to treat `stderr` as how the `stdout` was before, which was handled this way. I can go back to lossy though.

File: helix-term/src/commands.rs [hunk 2]
```diff
@@ -5701,27 +5701,28 @@ async fn shell_impl_async(
...
-            None => bail!("Shell command failed"),
+    let output = if !output.status.success() {
+        if output.stderr.is_empty() {
+            match output.status.code() {
+                Some(exit_code) => bail!("Shell command failed: status {}", exit_code),
+                None => bail!("Shell command failed"),
+            }
         }
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        log::error!("Shell error: {stderr}");
```
Comment (the-mikedavis) [inline helix-term/src/commands.rs:5712]: Now that we print this I wonder if we need to log an error anymore. I think we're safe to remove this line

File: helix-term/src/commands.rs [hunk 3]
```diff
@@ -5701,27 +5701,26 @@ async fn shell_impl_async(
...
-        log::debug!(
-            "Command printed to stderr: {}",
-            String::from_utf8_lossy(&output.stderr).to_string()
-        );
-    }
+        let stderr = String::from_utf8_lossy(&output.stderr);
+        log::debug!("Command printed to stderr: {stderr}");
+        stderr
+    } else {
+        let stdout = std::str::from_utf8(&output.stdout)
```
Comment (the-mikedavis) [inline helix-term/src/commands.rs:5718]: Like @archseer says above let's use the `from_ut8_lossy` here too. It's a change from the old code but I think we should be consistent and show something even when there's invalid UTF8

**Review by the-mikedavis — APPROVED** [review]

**Review by pascalkuthe — APPROVED** [review]
great QOL improvement!

**Pull request merged**